lib/hexpm_web/views/dashboard/audit_log_view.ex

```diff
@@ -6,117 +6,121 @@ defmodule HexpmWeb.Dashboard.AuditLogView do
+  def humanize_audit_log_info(%{action: "email.add"} = log) do
     "Add email #{log.params["email"]}"
   end
 
-  def humanize_audit_log_info(log = %{action: "email.remove"}) do
+  def humanize_audit_log_info(%{action: "email.remove"} = log) do
     "Remove email #{log.params["email"]}"
   end
 
-  def humanize_audit_log_info(log = %{action: "email.primary"}) do
+  def humanize_audit_log_info(%{action: "email.primary"} = log) do
     "Set email #{log.params["email"]} as primary email"
   end
 
-  def humanize_audit_log_info(log = %{action: "email.public"}) do
+  def humanize_audit_log_info(%{action: "email.public"} = log) do
     "Set email #{log.params["email"]} as public email"
   end
 
-  def humanize_audit_log_info(log = %{action: "email.gravatar"}) do
+  def humanize_audit_log_info(%{action: "email.gravatar"} = log) do
     "Set email #{log.params["email"]} as gravatar email"
   end
 
-  def humanize_audit_log_info(_log = %{action: "user.create"}) do
+  def humanize_audit_log_info(%{action: "user.create"} = _log) do
     "Create user account"
   end
 
-  def humanize_audit_log_info(_log = %{action: "user.update"}) do
+  def humanize_audit_log_info(%{action: "user.update"} = _log) do
     "Update user profile"
   end
 
-  def humanize_audit_log_info(log = %{action: "organization.create"}) do
+  def humanize_audit_log_info(%{action: "organization.create"} = log) do
     "Create organization #{log.params["name"]}"
   end
 
-  def humanize_audit_log_info(log = %{action: "organization.member.add"}) do
+  def humanize_audit_log_info(%{action: "organization.member.add"} = log) do
     "Add user #{log.params["user"]["username"]} to organization #{
       log.params["organization"]["name"]
     }"
   end
 
-  def humanize_audit_log_info(log = %{action: "organization.member.remove"}) do
+  def humanize_audit_log_info(%{action: "organization.member.remove"} = log) do
     "Remove user #{log.params["user"]["username"]} from organization #{
       log.params["organization"]["name"]
     }"
   end
 
-  def humanize_audit_log_info(log = %{action: "organization.member.role"}) do
+  def humanize_audit_log_info(%{action: "organization.member.role"} = log) do
     "Change user #{log.params["user"]["username"]}'s role to #{log.params["role"]} in organization #{
       log.params["organization"]["name"]
     }"
   end
 
-  def humanize_audit_log_info(_log = %{action: "password.reset.init"}) do
+  def humanize_audit_log_info(%{action: "password.reset.init"} = _log) do
     "Request to reset password"
   end
 
-  def humanize_audit_log_info(_log = %{action: "password.reset.finish"}) do
+  def humanize_audit_log_info(%{action: "password.reset.finish"} = _log) do
     "Reset password successfully"
   end
 
-  def humanize_audit_log_info(_log = %{action: "password.update"}) do
+  def humanize_audit_log_info(%{action: "password.update"} = _log) do
     "Update password"
   end
+
+  def humanize_audit_log_info(%{action: action} = _log) do
+    action
+  end
```

Comment:
The action code is internal so we should rather show a human version or not show it all.

Reply:
Hi @ericmj!

It's changed to return empty string in 4f914378d926faa79e97d8016e7b610bf861d0b7
(Not sure if it's what you mean by "not show it at all")

BTW do we have a monitor service so we can get notified when a recognized audit log occurs?

Reply:
> It's changed to return empty string in 4f91437
(Not sure if it's what you mean by "not show it at all")

It should not show up as a table entry at all.

> BTW do we have a monitor service so we can get notified when a recognized audit log occurs?

If we instead try to handle all audit logs then we would get a FunctionClauseError report in rollbar.

Reply:
If you instead want to support all audit logs without introducing a fallback you can compare the humanize function with the extract_params function: https://github.com/hexpm/hexpm/blob/master/lib/hexpm/accounts/audit_log.ex#L98

For example we seem to be missing all billing actions.

Reply:
IMO, a fallback is necessary in case of:
1. `humanize` function and `extract_params` function get unsynced (i.e. we forget to add a `humanize` handle after adding a new `action`)
2. an unexpected audit log action (e.g. a package should not have any `billing.*` actions)

> It should not show up as a table entry at all.

So I'd suggest to not show any unexpected audit logs and at the same time report it to rollbar:

```elixir
  def humanize_audit_log_info(%{action: "password.update"} = _log) do
    {:ok, "Update password"}
  end

  def humanize_audit_log_info(log) do
    Rollbax.report(:throw, :unexpected_audit_log, {__MODULE__, log.action})
    {:error, :unexpected_action}
  end

  # ...
  do not show the table row when humanize returns {:error, _}
```

This way, the user experience is better because the page won't crash due to `FunctionClauseError` and we still get the error report in rollbar.

The only thing I'm not sure is: is it okay to use `Rollbax.report(:throw)` like this? Or shall we define an custom exception and report that instead?

Reply:
The points you listed are both bugs and I don't like that kind of defensive programming that try to protect against possible future bugs, it is impossible to prevent them all so I think protecting against them only clutter the code base. The bugs would be pretty harmless, the audit log page would not render for the user (it is not critical that it does) and we would get a report in rollbar and fix it as soon as we can.

I think it would be better to implement the humanize function for all actions and leave it at that for now.

Reply:
> it is impossible to prevent them all so I think protecting against them only clutter the code base. The bugs would be pretty harmless, the audit log page would not render for the user (it is not critical that it does) and we would get a report in rollbar and fix it as soon as we can.

Agree. It was also why I didn't implement `PackageView.humanize_audit_log_info` functions for all actions.

humanizing `billing.*` audit logs for user is necessary, thank you for implementing that!